### ISSUE TITLE
Provide a clear follow up status for each resources at the end of the log

### ISSUE THREAD

**Issue by remi-f-artelia** [hashicorp/terraform#34448]
### Terraform Version

```shell
Terraform v1.6.2
```

### Use Cases

Following if each resources have been correctly `created`|`modified`|`deleted `can be tedious in case of deployment with a big amount of resources.
Moreover, if some of the action works but others fails it's sometimes difficult to find out which ones have worked and other not.

If you take the following log for instance, it's very difficult to follow up each resources status at the end of the apply

## Initial log (only the apply part)

https://gist.github.com/remi-f-artelia/483cce5f74cf11fd160c44d310d5029a

### Attempted Solutions

For now I just tried to parsed the text output with a powershell script for a one shot use case so my script is not robust enough to share it there.

I saw that it's possible to have a json output so maybe it would be the solution to avoid parsing the text like I did.
However I saw in these issues [34438](https://github.com/hashicorp/terraform/issues/34438), [33719](https://github.com/hashicorp/terraform/issues/33719) that these options do not provide json only and are not working for every commands.
And [the doc](https://developer.hashicorp.com/terraform/internals/json-format) says that's not possible to use it for `apply` or `destroy` commands (which would be my use case)

And the [official documentation](https://developer.hashicorp.com/terraform/cli/config/environment-variables) do not talk about these options (`TF_LOG_PROVIDER` or `TF_LOG_CORE`) so I'm confused on how to have the idea to use them...

**Edit**: I found the info [here](https://developer.hashicorp.com/terraform/internals/debugging) but it's not very clear on what these can offer

> Logging can be enabled separately for Terraform itself and the provider plugins using the TF_LOG_CORE or TF_LOG_PROVIDER environment variables. These take the same level arguments as TF_LOG, but only activate a subset of the logs.

### Proposal

Create a final output with a table showing clearly the action attempted per resource and the final status for this resource with at least the following keys

```json
{
  resource_type = <resource_type>
  resource_name = <resource_name>
  resource_id   = <resource_id>
  initial_state = <resource_state>
  final_state   = <final_state>
}
```

## Suggestion of output

https://gist.github.com/remi-f-artelia/004670c3c43120e04becbf2eecbb6254

**Comment by ART-jbfontaine** [comment]
Nice !

**Comment by crw** [comment]
Thanks for this feature request! If you are viewing this issue and would like to indicate your interest, please use the 👍 reaction on the issue description to upvote this issue. We also welcome additional use case descriptions. Thanks again!

**Comment by crw** [comment]
@remi-f-artelia No. It is possible to do what you describe, but would be a design decision the core team would need to make. There is some other functionality that is being developed that does stream full objects as it applies, so it may make sense in the future to accept this proposal to make the CLI overall more consistent. That said, there is no active work going on for this functionality.
